src/app/modules/node/sagas.ts

```diff
@@ -92,6 +100,80 @@ export function* handleCheckAuth(action: ReturnType<typeof actions.checkAuth>):
   });
 }
 
+export function* handleUpdateNodeUrl(action: ReturnType<typeof actions.updateNodeUrl>): SagaIterator {
+  try {
+    const newUrl = action.payload;
+    
+    // passowrd is needed to decrypt the admin macaroon
+    yield call(requirePassword);
+
+    // get current macaroons from state as its needed to store the new url
+    const { readonlyMacaroon } = yield select(selectSyncedUnencryptedNodeState);
+    const { adminMacaroon } = yield select(selectSyncedEncryptedNodeState);
+
+    // connect to the url to test if it's working
+    yield put(actions.checkNode(newUrl));
```

Comment:
This action sets `url` to null, and if the connection fails, it leaves it null in the reducer. This is fixed on a new session if it doesn't get synced, but if it did, it would bork your session.

Reply:
<img width="984" alt="screen shot 2019-01-10 at 11 54 14 pm" src="https://user-images.githubusercontent.com/649992/51014397-701e6180-1535-11e9-9393-4ba1ad1f0d8f.png">

Note the empty input on the left

Reply:
Ewwww.. great catch! Not sure how I missed that. I'll fix it.

Reply:
I fixed this by just calling `setNode` using the old url if the update fails.